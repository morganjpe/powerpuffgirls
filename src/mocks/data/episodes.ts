const episodes = [
  {
    id: 657308,
    url: 'https://www.tvmaze.com/episodes/657308/the-powerpuff-girls-1x01-escape-from-monster-island',
    name: 'Escape from Monster Island',
    season: 1,
    number: 1,
    type: 'regular',
    airdate: '2016-04-04',
    airtime: '18:00',
    airstamp: '2016-04-04T22:00:00+00:00',
    runtime: 11,
    image: {
      medium:
        'https://static.tvmaze.com/uploads/images/medium_landscape/53/132617.jpg',
      original:
        'https://static.tvmaze.com/uploads/images/original_untouched/53/132617.jpg',
    },
    summary:
      "<p>Bubbles wins two tickets to a concert and has to pick between Blossom and Buttercup to go with her. Meanwhile, the Mayor's plane goes down over Monster Island and it's up to the girls to rescue him.</p>",
    _links: {
      self: {
        href: 'https://api.tvmaze.com/episodes/657308',
      },
    },
  },
  {
    id: 657309,
    url: 'https://www.tvmaze.com/episodes/657309/the-powerpuff-girls-1x02-princess-buttercup',
    name: 'Princess Buttercup',
    season: 1,
    number: 2,
    type: 'regular',
    airdate: '2016-04-04',
    airtime: '18:00',
    airstamp: '2016-04-04T22:00:00+00:00',
    runtime: 15,
    image: {
      medium:
        'https://static.tvmaze.com/uploads/images/medium_landscape/53/132618.jpg',
      original:
        'https://static.tvmaze.com/uploads/images/original_untouched/53/132618.jpg',
    },
    summary:
      '<p>Buttercup becomes enamored with a team of roller derby girls called the Derbytantes. When she starts spending all of her time with them, Princess Morbucks seeks to be her replacement in the Powerpuff Girls.</p>',
    _links: {
      self: {
        href: 'https://api.tvmaze.com/episodes/657309',
      },
    },
  },
  {
    id: 1076814,
    url: 'https://www.tvmaze.com/episodes/1076814/the-powerpuff-girls-2x01-the-last-donnycorn',
    name: 'The Last Donnycorn',
    season: 2,
    number: 1,
    type: 'regular',
    airdate: '2017-03-03',
    airtime: '18:30',
    airstamp: '2017-03-03T23:30:00+00:00',
    runtime: 30,
    image: {
      medium:
        'https://static.tvmaze.com/uploads/images/medium_landscape/100/251774.jpg',
      original:
        'https://static.tvmaze.com/uploads/images/original_untouched/100/251774.jpg',
    },
    summary: null,
    _links: {
      self: {
        href: 'https://api.tvmaze.com/episodes/1076814',
      },
    },
  },
  {
    id: 1089606,
    url: 'https://www.tvmaze.com/episodes/1089606/the-powerpuff-girls-2x02-green-wing',
    name: 'Green Wing',
    season: 2,
    number: 2,
    type: 'regular',
    airdate: '2017-03-10',
    airtime: '18:45',
    airstamp: '2017-03-10T23:45:00+00:00',
    runtime: 15,
    image: {
      medium:
        'https://static.tvmaze.com/uploads/images/medium_landscape/101/254228.jpg',
      original:
        'https://static.tvmaze.com/uploads/images/original_untouched/101/254228.jpg',
    },
    summary: null,
    _links: {
      self: {
        href: 'https://api.tvmaze.com/episodes/1089606',
      },
    },
  },
  {
    id: 1454434,
    url: 'https://www.tvmaze.com/episodes/1454434/the-powerpuff-girls-3x01-not-so-secret-service',
    name: 'Not So Secret Service',
    season: 3,
    number: 1,
    type: 'regular',
    airdate: '2018-05-13',
    airtime: '16:45',
    airstamp: '2018-05-13T20:45:00+00:00',
    runtime: 15,
    image: {
      medium:
        'https://static.tvmaze.com/uploads/images/medium_landscape/156/390583.jpg',
      original:
        'https://static.tvmaze.com/uploads/images/original_untouched/156/390583.jpg',
    },
    summary:
      "<p>After Mojo Jojo's latest creation threatens to blast him into tiny bits, the Powerpuff Girls are forced to take Mojo in.</p>",
    _links: {
      self: {
        href: 'https://api.tvmaze.com/episodes/1454434',
      },
    },
  },
];

export default episodes;
