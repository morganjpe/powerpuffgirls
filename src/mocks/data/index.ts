export { default as episodes } from './episodes';
export { default as show } from './show';
export { default as search } from './search';
