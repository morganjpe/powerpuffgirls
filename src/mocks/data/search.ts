const search = [
  {
    score: 0.38040563,
    show: {
      id: 49892,
      url: 'https://www.tvmaze.com/shows/49892/the-powerpuff-girls',
      name: 'The Powerpuff Girls',
      type: 'Scripted',
      language: 'English',
      genres: [],
      status: 'In Development',
      runtime: null,
      averageRuntime: null,
      premiered: null,
      officialSite: null,
      schedule: {
        time: '',
        days: [],
      },
      rating: {
        average: null,
      },
      weight: 95,
      network: {
        id: 5,
        name: 'The CW',
        country: {
          name: 'United States',
          code: 'US',
          timezone: 'America/New_York',
        },
      },
      webChannel: null,
      dvdCountry: null,
      externals: {
        tvrage: null,
        thetvdb: null,
        imdb: 'tt12976292',
      },
      image: {
        medium:
          'https://static.tvmaze.com/uploads/images/medium_portrait/307/768002.jpg',
        original:
          'https://static.tvmaze.com/uploads/images/original_untouched/307/768002.jpg',
      },
      summary:
        '<p>The superhero children have grown and are now disillusioned twentysomethings who resent having lost their childhood to crime fighting. Will they agree to reunite now that the world needs them more than ever?</p>',
      updated: 1619036522,
      _links: {
        self: {
          href: 'https://api.tvmaze.com/shows/49892',
        },
      },
    },
  },
  {
    score: 0.37959835,
    show: {
      id: 6771,
      url: 'https://www.tvmaze.com/shows/6771/the-powerpuff-girls',
      name: 'The Powerpuff Girls',
      type: 'Animation',
      language: 'English',
      genres: ['Comedy', 'Action', 'Science-Fiction'],
      status: 'Running',
      runtime: 15,
      averageRuntime: 15,
      premiered: '2016-04-04',
      officialSite:
        'https://www.cartoonnetwork.com/video/powerpuff-girls/index.html',
      schedule: {
        time: '17:30',
        days: ['Sunday'],
      },
      rating: {
        average: 5.8,
      },
      weight: 86,
      network: {
        id: 11,
        name: 'Cartoon Network',
        country: {
          name: 'United States',
          code: 'US',
          timezone: 'America/New_York',
        },
      },
      webChannel: null,
      dvdCountry: null,
      externals: {
        tvrage: null,
        thetvdb: 307473,
        imdb: 'tt4718304',
      },
      image: {
        medium:
          'https://static.tvmaze.com/uploads/images/medium_portrait/60/151357.jpg',
        original:
          'https://static.tvmaze.com/uploads/images/original_untouched/60/151357.jpg',
      },
      summary:
        "<p>The city of Townsville may be a beautiful, bustling metropolis, but don't be fooled! There's evil afoot! And only three things can keep the bad guys at bay: Blossom, Bubbles and Buttercup, three super-powered little girls, known to their fans (and villains everywhere) as <b>The Powerpuff Girls</b>. Juggling school, bedtimes, and beating up giant monsters may be daunting, but together the Powerpuff Girls are up to the task. Battling a who's who of evil, they show what it really means to \"fight like a girl.\"</p>",
      updated: 1573685920,
      _links: {
        self: {
          href: 'https://api.tvmaze.com/shows/6771',
        },
        previousepisode: {
          href: 'https://api.tvmaze.com/episodes/1662010',
        },
      },
    },
  },
  {
    score: 0.37790465,
    show: {
      id: 1955,
      url: 'https://www.tvmaze.com/shows/1955/the-powerpuff-girls',
      name: 'The Powerpuff Girls',
      type: 'Animation',
      language: 'English',
      genres: ['Action', 'Children', 'Crime'],
      status: 'Ended',
      runtime: 30,
      averageRuntime: 30,
      premiered: '1998-11-18',
      officialSite: null,
      schedule: {
        time: '',
        days: ['Friday'],
      },
      rating: {
        average: 7.7,
      },
      weight: 0,
      network: {
        id: 11,
        name: 'Cartoon Network',
        country: {
          name: 'United States',
          code: 'US',
          timezone: 'America/New_York',
        },
      },
      webChannel: null,
      dvdCountry: null,
      externals: {
        tvrage: 6102,
        thetvdb: 76200,
        imdb: 'tt0175058',
      },
      image: {
        medium:
          'https://static.tvmaze.com/uploads/images/medium_portrait/11/27896.jpg',
        original:
          'https://static.tvmaze.com/uploads/images/original_untouched/11/27896.jpg',
      },
      summary:
        '<p><b>The Powerpuff Girls</b> is an award-winning animated action series about three super-powered little girls Bubbles, Blossom and Buttercup with one very big mission: saving the world before bedtime.</p>',
      updated: 1592706624,
      _links: {
        self: {
          href: 'https://api.tvmaze.com/shows/1955',
        },
        previousepisode: {
          href: 'https://api.tvmaze.com/episodes/160254',
        },
      },
    },
  },
  {
    score: 0.33375174,
    show: {
      id: 26437,
      url: 'https://www.tvmaze.com/shows/26437/demashitaa-powerpuff-girls-z',
      name: 'Demashitaa! Powerpuff Girls Z',
      type: 'Animation',
      language: 'Japanese',
      genres: ['Comedy', 'Action', 'Anime', 'Children'],
      status: 'Ended',
      runtime: 24,
      averageRuntime: 24,
      premiered: '2006-07-01',
      officialSite: null,
      schedule: {
        time: '',
        days: ['Saturday'],
      },
      rating: {
        average: null,
      },
      weight: 57,
      network: {
        id: 76,
        name: 'TV Tokyo',
        country: {
          name: 'Japan',
          code: 'JP',
          timezone: 'Asia/Tokyo',
        },
      },
      webChannel: null,
      dvdCountry: null,
      externals: {
        tvrage: null,
        thetvdb: 79472,
        imdb: 'tt0989771',
      },
      image: {
        medium:
          'https://static.tvmaze.com/uploads/images/medium_portrait/104/260281.jpg',
        original:
          'https://static.tvmaze.com/uploads/images/original_untouched/104/260281.jpg',
      },
      summary:
        '<p>Based on the American cartoon, "The Powerpuff Girls", Professor Utonium and his son Ken are studying the strange Chemical X. When a mochi cake falls into the formula, it changes into Chemical Z. Ken fires a ray of Chemical Z at a glacier in Tokyo Bay, however that causes strange black and white lights to scatter. The black lights turn its victims into monsters, with emphasis on a black light hitting a zoo monkey and transforming him into Mojo Jojo. The white lights, however, aim at three normal girls, Momoko, Miyako, and Kaoru. This gives them the power to transform into superheroes, the Powerpuff Girls Z!</p>',
      updated: 1573587031,
      _links: {
        self: {
          href: 'https://api.tvmaze.com/shows/26437',
        },
        previousepisode: {
          href: 'https://api.tvmaze.com/episodes/1112429',
        },
      },
    },
  },
];

export default search;
