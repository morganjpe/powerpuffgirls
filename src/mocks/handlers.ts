import { show, search, episodes } from './data';
import { rest } from 'msw';

const path = 'https://api.tvmaze.com/';

const handlers = [
  rest.get(`${path}shows/6771`, (req, res, ctx) =>
    res(ctx.status(200), ctx.json(show))
  ),
  rest.get(`${path}shows/49892`, (req, res, ctx) =>
    res(ctx.status(200), ctx.json(show))
  ),
  rest.get(`${path}shows/6771/episodes`, (req, res, ctx) =>
    res(ctx.status(200), ctx.json(episodes))
  ),
  rest.get(`${path}shows/49892/episodes`, (req, res, ctx) =>
    res(ctx.status(200), ctx.json(episodes))
  ),
  rest.get(`${path}search/*`, (req, res, ctx) =>
    res(ctx.status(200), ctx.json(search))
  ),
];

export default handlers;
