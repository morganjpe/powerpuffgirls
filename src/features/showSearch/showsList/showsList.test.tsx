import { screen } from '@testing-library/react';
import ShowsList from './showsList';
import renderWithBrowser from '../../../testutils';
import pretty from 'pretty';

import { search } from '../../../mocks/data';

describe('<ShowsList /> component', () => {
  it('should display no shows if no shows have been found', () => {
    const { container } = renderWithBrowser(
      <ShowsList isLoading={false} search="efqedqwdqwd" shows={[]} />
    );
    screen.getByText(/could not find shows/i);
    expect(pretty(container.innerHTML)).toMatchSnapshot();
  });

  it('should display a list of show cards', () => {
    const { container } = renderWithBrowser(
      <ShowsList isLoading={false} search="dogs" shows={search} />
    );
    screen.getAllByText(/The Powerpuff Girls/i);
    screen.getByText(/Demashitaa! Powerpuff Girls Z/i);
    expect(pretty(container.innerHTML)).toMatchSnapshot();
  });
});
