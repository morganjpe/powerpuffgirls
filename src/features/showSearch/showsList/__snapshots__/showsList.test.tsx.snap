// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ShowsList /> component should display a list of show cards 1`] = `
"<div class=\\"sc-hKFxyN cuWGjR\\">
  <div>
    <ul>
      <li>
        <div><a href=\\"/show/49892\\">
            <div class=\\"sc-gtsrHT jMaFal\\"><img src=\\"https://static.tvmaze.com/uploads/images/medium_portrait/307/768002.jpg\\" alt=\\"The Powerpuff Girls\\" class=\\"sc-dlnjwi ejJcht\\"><img class=\\"main\\" tw=\\"rounded\\" src=\\"https://static.tvmaze.com/uploads/images/medium_portrait/307/768002.jpg\\" alt=\\"The Powerpuff Girls\\"></div>
            <h4>The Powerpuff Girls</h4>
          </a></div>
      </li>
      <li>
        <div><a href=\\"/show/6771\\">
            <div class=\\"sc-gtsrHT jMaFal\\"><img src=\\"https://static.tvmaze.com/uploads/images/medium_portrait/60/151357.jpg\\" alt=\\"The Powerpuff Girls\\" class=\\"sc-dlnjwi ejJcht\\"><img class=\\"main\\" tw=\\"rounded\\" src=\\"https://static.tvmaze.com/uploads/images/medium_portrait/60/151357.jpg\\" alt=\\"The Powerpuff Girls\\"></div>
            <h4>The Powerpuff Girls</h4>
          </a></div>
      </li>
      <li>
        <div><a href=\\"/show/1955\\">
            <div class=\\"sc-gtsrHT jMaFal\\"><img src=\\"https://static.tvmaze.com/uploads/images/medium_portrait/11/27896.jpg\\" alt=\\"The Powerpuff Girls\\" class=\\"sc-dlnjwi ejJcht\\"><img class=\\"main\\" tw=\\"rounded\\" src=\\"https://static.tvmaze.com/uploads/images/medium_portrait/11/27896.jpg\\" alt=\\"The Powerpuff Girls\\"></div>
            <h4>The Powerpuff Girls</h4>
          </a></div>
      </li>
      <li>
        <div><a href=\\"/show/26437\\">
            <div class=\\"sc-gtsrHT jMaFal\\"><img src=\\"https://static.tvmaze.com/uploads/images/medium_portrait/104/260281.jpg\\" alt=\\"Demashitaa! Powerpuff Girls Z\\" class=\\"sc-dlnjwi ejJcht\\"><img class=\\"main\\" tw=\\"rounded\\" src=\\"https://static.tvmaze.com/uploads/images/medium_portrait/104/260281.jpg\\" alt=\\"Demashitaa! Powerpuff Girls Z\\"></div>
            <h4>Demashitaa! Powerpuff Girls Z</h4>
          </a></div>
      </li>
    </ul>
  </div>
</div>"
`;

exports[`<ShowsList /> component should display no shows if no shows have been found 1`] = `
"<div class=\\"sc-hKFxyN cuWGjR\\">
  <div>Could not find shows</div>
</div>"
`;
